### DISCUSSION TITLE
Static generated html pages and cache control

### DISCUSSION THREAD

**Discussion by derfoufi** [vercel/next.js#18418] [Help]
Hi

First of all, thank you to the community for all the work done which allowed this framework to be what it is today.

In the official doc i found this :
- _"Static Generation (Recommended): The HTML is generated at build time and will be reused on each request."_
- _"Statically generated pages can be cached by CDN with no extra configuration to boost performance"_

The statics js files corresponding to the pages have _"Cache-Control: public, max-age=31536000, immutable"_ and was cached by a cdn.
But the static generated html pages have no cache control and are not cached by a cdn.

Can some one please explain me what i missed ?

Thank you

**Comment by MathiasKandelborg** [comment]
There's a lot of things going on. What are you looking to get answered? How has Next.js implemented caching strategies, if you've found undesired behavior, or something else? 

Next.js does a lot of things under the hood to be more performant and efficient. I suggest looking through the source code to get a clearer picture of exactly what's being done.

As a reference, here's my [still developing website](https://true-framework.vercel.app/), which uses next-pwa to have offline support. Look at the network tab and check out how much data is transferred on reloads/navigations after the first page-load.
That's all out-of-the-box caching and next-pwa; basically, plug & play. I could get more aggressive or relaxed depending on what is on a given page - e.g., sensitive information - but haven't touched those areas yet.

**Comment by derfoufi** [comment]
I just want to know how can i make my static page cached by a cdn.
here is an [example](https://deco-exemple.web.app) of a simple NextJs app and hosted in firebase, it run in google function.
My problem is that my static home page is served without cache control as you can see at the network tab, so it isn't cached by cdn.

**Reply by MathiasKandelborg** [reply]
It is done automatically by Vercel it seems.

If you want to optimize the cache for Firebase and google functions, I've found some resources for that platform:

https://firebase.google.com/docs/hosting/manage-cache

https://firebase.google.com/docs/functions/tips

**Reply by thanhtoan1196** [reply]
@MathiasKandelborg @derfoufi I also have the same question, my page is not cached by cloudflare either
https://test11zz.com/doc-truyen/du-tinh-loi-moi-cua-boss-than-bi

**Comment by leerob** [comment]
It's now possible to customize `cache-control` as of v14.2.10. Please upgrade and give it a try!

https://github.com/vercel/next.js/pull/69802
